### PR TITLE
bug fixes in ST recipes

### PR DESCRIPTION
**Pull request by chintu619**
* Change sampling frequency in `fbank.conf` and `pitch.conf` in Covost2 recipe
* In `run.sh`, if language is low resource, then have more speed perturbations. Fix typos for test sets
* In `st.sh`
  * Fixed directory naming issues to avoid replacement for different language pairs
  * Replaced `>>` with `>` to replace previous inference results
  * Fixed removal of empty text in stage 4
  * When removing utterance-ID from `ref.trn.org` and `hyp.trn.org`, the current implementation removes all words in parenthesis instead of removing just the utterance-ID from the end of each line. Fixed this by changing `perl -pe 's/\([^\)]+\)//g;'` to `perl -pe 's/\([^\)]+\)$//g;'`
  * Typo fix: `suuport` to `support`

### CODE REVIEW

**Comment by sw005320** [comment]
@chintu619, can you reflect @ftshijt’s comments?

File: egs2/TEMPLATE/st1/st.sh [hunk 1]
```diff
@@ -717,8 +695,10 @@ if ! "${skip_data_prep}"; then
             fi
 
             # Remove empty text
-            <"${data_feats}/org/${dset}/text" \
-                awk ' { if( NF != 1 ) print $0; } ' >"${data_feats}/${dset}/text"
+            for utt_extra_file in ${utt_extra_files}; do
+                <"${data_feats}/org/${dset}/${utt_extra_file}" \
```
[the remark this answers is not in the document]
Comment (chintu619) [reply egs2/TEMPLATE/st1/st.sh:699]: Can `fix_data_dir.sh` handle empty text lines? I thought it only checked for non-empty audio files (either duration > 0 or num_frames > 0).

File: egs2/TEMPLATE/st1/st.sh [hunk 2]
```diff
@@ -1485,31 +1465,31 @@ if ! "${skip_eval}"; then
...
-            perl -pe 's/\([^\)]+\)//g;' "${_scoredir}/hyp.trn.org" > "${_scoredir}/hyp.trn"
+            perl -pe 's/\([^\)]+\)$//g;' "${_scoredir}/ref.trn.org" > "${_scoredir}/ref.trn"
+            perl -pe 's/\([^\)]+\)$//g;' "${_scoredir}/hyp.trn.org" > "${_scoredir}/hyp.trn"
 
             # detokenizer
             detokenizer.perl -l ${tgt_lang} -q < "${_scoredir}/ref.trn" > "${_scoredir}/ref.trn.detok"
             detokenizer.perl -l ${tgt_lang} -q < "${_scoredir}/hyp.trn" > "${_scoredir}/hyp.trn.detok"
 
             if [ ${tgt_case} = "tc" ]; then
-                echo "Case sensitive BLEU result (single-reference)" >> ${_scoredir}/result.tc.txt
```
[the remark this answers is not in the document]
Comment (chintu619) [reply egs2/TEMPLATE/st1/st.sh:1496]: The original format was appending latest scores towards the end of result files. Therefore the `RESULTS.md` file was not updating with latest results. As suggested, I followed the same strategy as `train.log` using a new pyscripts utility: [rotate_logfile.py](egs2/TEMPLATE/asr1/pyscripts/utils/rotate_logfile.py). Should I add a unit test for this utility?

File: egs2/TEMPLATE/st1/st.sh [hunk 3]
```diff
@@ -1485,31 +1465,31 @@ if ! "${skip_eval}"; then
...
+                echo "Case sensitive BLEU result (single-reference)" > ${_scoredir}/result.tc.txt
                 sacrebleu "${_scoredir}/ref.trn.detok" \
                           -i "${_scoredir}/hyp.trn.detok" \
                           -m bleu chrf ter \
-                          >> ${_scoredir}/result.tc.txt
```
[the remark this answers is not in the document]
Comment (chintu619) [reply egs2/TEMPLATE/st1/st.sh:1500]: Sorry, my mistake. Reverted.

File: egs2/TEMPLATE/st1/st.sh [hunk 4]
```diff
@@ -1532,25 +1512,25 @@ if ! "${skip_eval}"; then
...
-                    perl -pe 's/\([^\)]+\)//g;' "${_scoredir}/ref.trn.org.${ref_idx}" > "${_scoredir}/ref.trn.${ref_idx}"
+                    perl -pe 's/\([^\)]+\)$//g;' "${_scoredir}/ref.trn.org.${ref_idx}" > "${_scoredir}/ref.trn.${ref_idx}"
                     detokenizer.perl -l ${tgt_lang} -q < "${_scoredir}/ref.trn.${ref_idx}" > "${_scoredir}/ref.trn.detok.${ref_idx}"
                     remove_punctuation.pl < "${_scoredir}/ref.trn.detok.${ref_idx}" > "${_scoredir}/ref.trn.detok.lc.rm.${ref_idx}"
                     case_sensitive_refs="${case_sensitive_refs} ${_scoredir}/ref.trn.detok.${ref_idx}"
                     case_insensitive_refs="${case_insensitive_refs} ${_scoredir}/ref.trn.detok.lc.rm.${ref_idx}"
                 done
 
                 if [ ${tgt_case} = "tc" ]; then
-                    echo "Case sensitive BLEU result (multi-references)" >> ${_scoredir}/result.tc.txt
```
[the remark this answers is not in the document]
Comment (chintu619) [reply egs2/TEMPLATE/st1/st.sh]: Again, my mistake. Reverted.

File: egs2/TEMPLATE/st1/st.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (chintu619) [reply egs2/TEMPLATE/st1/st.sh:699]: As we don’t need to have a `text` file for ST recipes, I removed the following line and only checked for empty lines in  `text.tc.src, text.tc.tgt, ...`.
```bash
   <"${data_feats}/org/${dset}/text" \
      awk ' { if( NF != 1 ) print $0; } ' >"${data_feats}/${dset}/text"
```

**Comment by ftshijt** [comment]
Hi @chintu619 , many thanks for the update! I will merge the current PR since it indeed help some of the logistics in ST. But could you further add test functions for the next scripts in a later PR?

**Pull request merged**